perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -44,13 +46,52 @@ enum Flags
     Flag_count
 };
 
-static void SetFlags(u16 Result, bool* FlagArray)
-{
-    FlagArray[Flag_SF] = Result & 0x8000;
-    FlagArray[Flag_ZF] = (Result == 0);
-}
+static void SetFlags(const instruction& Instruction, const u16& LeftOperandValue, 
+    const u16& RightOperandValue, const u16& Result, bool* FlagArray)
+{   
+    size_t HighOrderBit = (Instruction.Operands[1].Register.Count == 1) ? 0x80 : 0x8000;
+    bool Parity = true;
+       
+    switch (Instruction.Op)
+    {
+        case Op_add: [[fallthrough]];
+        case Op_sub: 
+        {
+            if (Instruction.Op == Op_add)
```

Comment:
You should put this in the case above, instead of using an if here. Kind of defeats the purpose of using a switch statement in the first place.

Reply:
I wanted to use fallthrough because the setting of SF, ZF and PF is the same in all 3 cases. Now that I think about it it's not a good reason to do that.
Fixed now with repeated code because I think it's easier to read than figuring out why there's only a fallthrough between sub and default, but I'm open to change it
8e816268a11dff001c4cd953721a8266e52f5822

Reply:
If the code is repeated that would be a good reason to use a function here instead.

Sure, it's not 3 usages, but there's another issue which makes it good to have a function. You ended up with two copies of code which does non trivial things. So it just makes things hard to read or parse visually. When I go to look at this code written by someone who is not me, I have no idea at first that it's the same code twice, so I have to read it two times anyway to confirm to myself that it does the same thing.

Repeating stuff is fine when it's not too complex and can be read rather easily. But with bitwise operations and stuff I don't recommend it.

Reply:
Another alternative is to not use a switch statement.

Reply:
It turns out I had a bug. I thought cmp didn't set all flags but it actually does.
The flag setting for sub and cmp is actually the same, so the switch statement works out well.
1f46ad7f442b0b8f36330045ff2877279d1e1639